sympy/functions/special/bessel.py

```diff
@@ -216,11 +217,44 @@ def _eval_rewrite_as_bessely(self, nu, z, **kwargs):
     def _eval_rewrite_as_jn(self, nu, z, **kwargs):
         return sqrt(2*z/pi)*jn(nu - S.Half, self.argument)
 
+    def _eval_as_leading_term(self, x, cdir=0):
+        nu, z = self.args
+        arg = z.as_leading_term(x)
+        if x in arg.free_symbols:
+            return z**nu
+        else:
+            return self.func(*self.args)
+
     def _eval_is_extended_real(self):
         nu, z = self.args
         if nu.is_integer and z.is_extended_real:
             return True
 
+    def _eval_nseries(self, x, n, logx, cdir=0):
+        from sympy.series.order import Order
+        nu, z = self.args
+
+        # In case of powers less than 1, number of terms need to be computed
+        # separately to avoid repeated callings of _eval_nseries with wrong n
+        try:
+            _, exp = z.leadterm(x)
+        except (ValueError, NotImplementedError):
+            return self
+
+        if exp.is_positive:
+            newn = ceiling(n/exp)
+            r = z._eval_nseries(x, n, logx, cdir).removeO()
+            t = (z**2)._eval_nseries(x, n, logx, cdir).removeO()
+
+            term = 1/gamma(nu + 1)
```

Comment:
If `nu` is a negative integer, then this is 0. We can use the equality `besselj(-n, x) = (-1)**n*besselj(n, x)`. So something like this:
```
            if nu.is_integer and nu.is_negative:
                nu = -nu
                term = (-1)**nu/factorial(nu)
            else:
                term = 1/gamma(nu + 1)
```

Reply:
`nu` cannot be a negative integer. It is handled in `eval` method - https://github.com/sympy/sympy/blob/b564d9ba2705ee2978766e1ee2102750834df68b/sympy/functions/special/bessel.py#L187-L188

Reply:
This does not mean that a symbolic nu were negative. But it will work for numerical values and that will probably suffice.